src/lib/controllers/TaskController.ts

```diff
@@ -0,0 +1,143 @@
+import { Response, NextFunction } from 'express';
+import { IRequest } from '@typings';
+import { logRequest, unauthorizedError } from '@utils';
+import { Task } from '@models';
+import { ModelController } from '@controllers';
+import { FORBIDDEN_FIELDS } from '@config';
+
+class TaskController extends ModelController<typeof Task> {
+    constructor() {
+        super(Task);
+    }
+
+    @logRequest
+    async create(
+        req: IRequest,
+        res: Response,
+        next: NextFunction
+    ): Promise<void | Response> {
+        try {
+            const { owner } = req;
+            const {
+                title,
+                description,
+                deadline,
+                dateBegin,
+                dateEnd,
+            } = req.body;
+
+            const newTask = await Task.create({
+                title,
+                description,
+                deadline,
+                dateBegin,
+                dateEnd,
+            });
+            await newTask.addUser(owner);
+            return res.status(201).json(newTask);
+        } catch (err) {
+            return next(err);
+        }
+    }
+
+    @logRequest
+    async addTaskUser(
+        req: IRequest,
+        res: Response,
+        next: NextFunction
+    ): Promise<void | Response> {
+        try {
+            const { user, task } = req;
+            const taskUsers = await task.getUsers();
+            /**
+             * Check if the invited user has already request to join the task
+             */
+            const userInTask = taskUsers.find(
```

Comment:
Juste tu peux m'éclairer sur ce que tu veux dire par "Check if the invited user has already request to join the task", tu veux dire qu'il y a un statut pending ?
Pour rappel on checkait ça au niveau de la ManyToMany Team/User parce que il y avait un statut pending